hyperspy/models/edsmodel.py

```diff
@@ -284,7 +286,8 @@ def add_polynomial_background(self, order=6):
         order: int
             The order of the polynomial
         """
-        background = create_component.Polynomial(order=order)
+        with ignore_warning(message="The definition of the `Polynomial`"):
+            background = create_component.Polynomial(order=order)
```

Comment:
By default, it will use the deprecated polynomial and it should use the new one by setting `legacy=False`.